src/userent.c

```diff
@@ -304,28 +305,35 @@ int pass_set(struct userrec *u, struct user_entry *e, void *buf)
         *p = '?';
       p++;
     }
-    if (u->flags & USER_BOT)
+    if ((u->flags & USER_BOT) || pass[0] == '+') {
       strlcpy(new, pass, sizeof new);
-      /* We do not need PASS2 for bots */
-    else if (pass[0] == '+') {
-      strlcpy(new, pass, sizeof new);
-      new2 = NULL; /* due to module api encrypted pass2 cannot be available here
-                    * Caller must do set_user(&USERENTRY_PASS2, u, password);
-                    * Probably only share.c:dup_userlist()
-                    */
+      /* USER_BOT: we do not need PASS2 for bots
+       * '+': due to module api encrypted pass2 cannot be available here
+       * caller must do set_user(&USERENTRY_PASS2, u, password);
+       * probably only share.c:dup_userlist()
+       */
     }
     else {
-      if (encrypt_pass)
-        encrypt_pass(pass, new);
+      if (encrypt_pass && (!encrypt_pass2 || !remove_pass))
+          encrypt_pass(pass, new);
       if (encrypt_pass2)
         new2 = encrypt_pass2(pass);
     }
-    if (encrypt_pass) {
+    if ((u->flags & USER_BOT) || (encrypt_pass &&
```

Comment:
@thommey what's the verdict here?

Reply:
before and after introduction of pbkdf2: bot passwords are stored in cleartext. The code doing this before pbkdf2 is here:  https://github.com/eggheads/eggdrop/blob/5182a229e7042a75fcf2c3c5bbf9835a90bad90d/src/userent.c#L259. when i added pbkdf2 i did a mistake in my initial PR. If blowfish was loaded,  in my initial pbkd2 PR there was a bug, preventing a bot from saving a bot password if blowfish mod is not laded. This could be the case if only pbkdf2 mod is loaded. The Error line is this one: https://github.com/eggheads/eggdrop/pull/1055/files#diff-dd198bb0cb017512db4c27ee75144ae992e8c4c963b8d3c25b85dbdf785a7263R323.